src/GovUk.Education.ExploreEducationStatistics.Publisher.Tests/Services/NotificationsServiceTests.cs

```diff
@@ -0,0 +1,201 @@
+            {
+                var notificationsService = BuildNotificationsService(
+                    contentDbContext, storageQueueService.Object);
+
+                await notificationsService.NotifySubscribersIfApplicable(release1.Id, release2.Id, release3.Id);
+
+                storageQueueService.Verify(mock => mock.AddMessages(
+                    PublicationQueue,
+                new List<PublicationNotificationMessage>
+                    {
+                        new()
+                        {
+                            Name = release1.Publication.Title,
+                            PublicationId = release1.Publication.Id,
+                            Slug = release1.Publication.Slug,
+                        },
+                        new()
+                        {
+                            Name = release3.Publication.Title,
+                            PublicationId = release3.Publication.Id,
+                            Slug = release3.Publication.Slug,
+                        },
+                    }), Times.Once);
+            }
+
+            MockUtils.VerifyAllMocks(storageQueueService);
+        }
+
+        [Fact]
+        public async Task NotifySubscribersIfApplicable_NotifyOnSet()
+        {
+            var release = new Release
+            {
+                Id = Guid.NewGuid(),
+                Publication = new Publication
+                {
+                    Id = Guid.NewGuid(),
+                    Title = "pub title",
+                    Slug = "pub-slug",
+                },
+                ReleaseStatuses = new List<ReleaseStatus>
+                {
+                    new()
+                    {
+                        Id = Guid.NewGuid(),
+                        ApprovalStatus = ReleaseApprovalStatus.Approved,
+                        Created = DateTime.UtcNow,
+                        NotifySubscribers = true,
+                        NotifiedOn = DateTime.UtcNow,
+                    },
+                },
+            };
+            var contentDbContextId = Guid.NewGuid().ToString();
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                await contentDbContext.AddRangeAsync(release);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                var notificationsService = BuildNotificationsService(contentDbContext);
+
+                await notificationsService.NotifySubscribersIfApplicable(release.Id);
+
+                // No interaction with storage queue service expected as NotifyOn is set
+                // in ReleaseStatus.
+            }
+        }
+
+        [Fact]
+        public async Task NotifySubscribersIfApplicable_NoReleaseIds()
+        {
+            var contentDbContextId = Guid.NewGuid().ToString();
```

Comment:
Minor one but you don't need an id here, a unique db name will be generated if the id is omitted.